spring-cloud-services-spring-connector/src/main/java/io/pivotal/spring/cloud/service/config/VaultTokenRenewer.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.pivotal.spring.cloud.service.config;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.cloud.config.client.ConfigClientProperties;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.HttpHeaders;
+import org.springframework.scheduling.annotation.Scheduled;
+import org.springframework.security.oauth2.client.OAuth2RestTemplate;
+import org.springframework.web.client.ResourceAccessException;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * Configuration for a periodic Vault token renewer. Conditionally configured if there
+ * is a {@link ConfigClientProperties} bean and if there is a `spring.cloud.config.token`
+ * property set.
+ * 
+ * By default, the token is renewed every 60 seconds and is renewed with a 5 minute time-to-live.
+ * The renew rate can be configured by setting `vault.token.renew.rate` to some value that is the 
+ * renewal rate in milliseconds. The renewal time-to-live can be specified with by setting
+ * `vault.token.ttl` to some value indicating the time-to-live in seconds.
+ * 
+ * @author cwalls
+ */
+@Configuration
+@ConditionalOnBean(ConfigClientProperties.class)
+@ConditionalOnProperty(name="spring.cloud.config.token")
+public class VaultTokenRenewer {
```

Comment:
Your point is well taken, but...I went back and forth on that for several hours yesterday. Technically it's only a configuration class because it's what ends up in `spring.factories` under `org.springframework.boot.autoconfigure.EnableAutoConfiguration`. Ideally I wanted to extract the guts of it into a component class called `VaultTokenRenewer`, but when I did that refactoring, there wasn't anything left in the configuration class.

In short, this is a component class (maybe had I annotated it with `@Component` that would've been more obvious--I'm not sure if that would've even worked, though). But it's treated like a `@Configuration` class for the sake of auto-configuration. 

But yeah...you're only asking for a class name change. I suppose that if this moves things forward then I can do that.